src/tools/execution-results.h

```diff
@@ -87,15 +87,22 @@ struct LoggingExternalInterface : public ShellExternalInterface {
 // we can only get results when there are no imports. we then call each method
 // that has a result, with some values
 struct ExecutionResults {
+  enum FunctionResultKind { Normal, Trap, Exception };
   struct FunctionResult {
     Literals values;
-    bool exception; // Whether an exception is uncaught and the function crashes
+    FunctionResultKind kind;
```

Comment:
Since the `values` are only meaningful if the `FunctionResultKind` is `Normal`, wdyt of using a `std::variant<Literals, Trap, Exception>` as `FunctionResult`?

Reply:
`Trap` and `Exception` are enum values and not types though..?

Reply:
You would also have to change them from being enum variants to being types:

```
struct Trap {};
struct Exception {};
using FunctionResult = std::variant<Literals, Trap, Exception>;
```

Reply:
Done